macOS tests fail with "STARTTLS upgrade failed: connection closed via error"

Log is here:
https://github.com/deltachat/deltachat-core-rust/actions/runs/4471878171/jobs/7860619826

The runner is macOS 12.6.3. We have successful CI runs on exactly the same CI runner version, so the issue is not due to update on GitHub actions.

Full error is:
```
  1) Integration tests
       configure:
     AssertionError: expected promise to be fulfilled but it was rejected with 'Error: Error:\n\n“STARTTLS upgrade failed: connection closed via error”'
```

The `Error:\n\n` part comes from `stock_str::configuration_failed` and `STARTTLS upgrade failed: connection closed via error` is the error returned by `inner_configure` and formatted with `{err:#}`.

We have not changed anything related to TLS on the core recently, but have switched from x.testrun.org to testrun.org and there is the same error on @Simon-Laux macos laptop.

The `connection closed via error` is an `errSSLClosedAbort 			= -9806,	/* connection closed via error */` error from Apple's security-framework.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -22,6 +22,7 @@
 - Delete expired messages using multiple SQL requests. #4158
 - Do not emit "Failed to run incremental vacuum" warnings on success. #4160
 - Ability to send backup over network and QR code to setup second device #4007
+- Disable buffering during STARTTLS setup. #4190
 
 ## [1.111.0] - 2023-03-05
 

--- a/src/smtp.rs
+++ b/src/smtp.rs
@@ -143,7 +143,7 @@ impl Smtp {
         // Run STARTTLS command and convert the client back into a stream.
         let client = smtp::SmtpClient::new().smtp_utf8(true);
         let transport = SmtpTransport::new(client, BufStream::new(socks5_stream)).await?;
-        let tcp_stream = transport.starttls().await?;
+        let tcp_stream = transport.starttls().await?.into_inner();
         let tls_stream = wrap_tls(strict_tls, hostname, tcp_stream)
             .await
             .context("STARTTLS upgrade failed")?;
@@ -199,7 +199,7 @@ impl Smtp {
         // Run STARTTLS command and convert the client back into a stream.
         let client = smtp::SmtpClient::new().smtp_utf8(true);
         let transport = SmtpTransport::new(client, BufStream::new(tcp_stream)).await?;
-        let tcp_stream = transport.starttls().await?;
+        let tcp_stream = transport.starttls().await?.into_inner();
         let tls_stream = wrap_tls(strict_tls, hostname, tcp_stream)
             .await
             .context("STARTTLS upgrade failed")?;

